hackage2nix/Main.hs

```diff
@@ -36,13 +35,30 @@ import           Distribution.Pretty                      ( prettyShow
                                                           )
 import           Distribution.Types.PackageName           ( PackageName )
 import           Distribution.Types.Version               ( Version )
+import           Nix                                      ( (@@)
+                                                          , mkSym
+                                                          , mkInt
+                                                          , mkStr
+                                                          , NExpr
+                                                          , ($=)
+                                                          , mkNonRecSet
+                                                          )
 import           Nix.Pretty                               ( prettyNix )
 import           System.Directory                         ( createDirectoryIfMissing
                                                           )
 import           System.Environment                       ( getArgs )
 import           System.FilePath                          ( (</>)
                                                           , (<.>)
                                                           )
+import Data.Char (isUpper)
+
+-- Avoid issues with case insensitive file systems by escaping upper case
+-- characters with a leading _ character.
+escapeUpperCase :: String -> String
```

Comment:
This sounds a bit like a custom hack to me … what about just use something like the base64 representation of strings we want to escape?

Reply:
As this is for filenames, base64 encoding would make it a lot less readable :-/

I wonder if we'd run into camelCase issues.

That would be converted into camel_case, and thus clash with a package named camel_case? Can we rule out that this would happen?

Reply:
I wanted the file names to still be human readable.  I guess we could include both human and base64 encoding, but then it is kind of a custom format again.

Reply:
My bad, I was thinking about case-insensitive base32!

Reply:
My hot take would be that here we had a good opportunity to hide this hack behind an abstraction, meaning a small lib. This is a micro Haskell project I would be glad to publish on Hackage!